govcd/security_tags.go

```diff
@@ -0,0 +1,172 @@
+package govcd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"net/url"
+)
+
+// GetSecurityTaggedEntities Retrieves the list of entities that have at least one tag assigned to it.
+// Besides, entityType, additional supported filters are:
+//   - tag - The tag to search by. I.e: filter=(tag==Web;entityType==vm)
+// This function works from API v36.1 (VCD 10.3.1+)
+func (vcdClient *VCDClient) GetSecurityTaggedEntities(filter string) ([]*types.SecurityTaggedEntity, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointSecurityTags
+	apiVersion, err := vcdClient.Client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := vcdClient.Client.OpenApiBuildEndpoint(endpoint, "/entities")
+	if err != nil {
+		return nil, err
+	}
+
+	queryParams := url.Values{}
+	queryParams.Set("filter", filter)
+	rawValues, err := vcdClient.Client.openApiGetAllPages(apiVersion, urlRef, queryParams, nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	securityTaggedEntities := make([]*types.SecurityTaggedEntity, len(rawValues))
+
+	for i, k := range rawValues {
+		var temp types.SecurityTaggedEntity
+		err = json.Unmarshal(k, &temp)
+		if err != nil {
+			return nil, fmt.Errorf("error when unmarshalling SecurityTaggedEntity - %s", err)
+		}
+		securityTaggedEntities[i] = &temp
+	}
+	return securityTaggedEntities, nil
+}
+
+// GetSecurityTagValues Retrieves the list of security tags that are in the organization and can be reused to tag an entity.
+// The list of tags include tags assigned to entities within the organization.
+// This function works from API v36.1 (VCD 10.3.1+)
+func (org *Org) GetSecurityTagValues(filter string) ([]*types.SecurityTagValue, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointSecurityTags
+	apiVersion, err := org.client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := org.client.OpenApiBuildEndpoint(endpoint, "/values")
+	if err != nil {
+		return nil, err
+	}
+
+	// If sysadmin, getting the org context and this method only works for org users
+	orgContextHeaders := make(map[string]string)
+	if org.client.IsSysAdmin {
+		orgContextHeaders["X-VMWARE-VCLOUD-AUTH-CONTEXT"] = org.Org.Name
```

Comment:
```suggestion
		orgContextHeaders[types.HeaderAuthContext] = org.Org.Name
		orgContextHeaders[types.HeaderTenantContext] = org.Org.ID
```

Reply:
Ok, followed what @Didainius suggested to add these headers :)